### DISCUSSION TITLE
not working inlay_hints

### DISCUSSION THREAD

**Discussion by scippio** [neovim/neovim#28936] [Q&A] [answered]
When I enable them by: `vim.lsp.inlay_hint.enable(true,nil) ` or `vim.lsp.inlay_hint.enable(true)` nothing is change. I'm run this command for check: `vim.notify(tostring(vim.lsp.inlay_hint.is_enabled()))` and it's return true ... but I see nothing (in my lua, php, go files) ...

Am I doing something wrong? Sorry.

I have version 0.10.0 from Arch package manager.

**Comment by frolvanya** [comment, marked as answer]
You have to [enable](https://www.reddit.com/r/neovim/comments/14em0f8/how_to_use_the_new_lsp_inlay_hints/) them in your lspconfig

Like this:
```lua
require("lspconfig").lua_ls.setup({
    settings = {
        Lua = {
            hint = {
                enable = true,
            }
        }
    }
})
require("lspconfig").gopls.setup({
    settings = {
        gopls = {
            hints = {
                assignVariableTypes = true,
                compositeLiteralFields = true,
                compositeLiteralTypes = true,
                constantValues = true,
                functionTypeParameters = true,
                parameterNames = true,
                rangeVariableTypes = true,
            },
        },
    },
})
```

You also can use [this](https://github.com/MysticalDevil/inlay-hints.nvim) to make things easier

P.S: Keep in mind that not all LSPs support inlay hints

**Comment by scippio** [comment]
Thank you! Is there some LSP for PHP with this support?

**Reply by frolvanya** [reply]
No, as far as I know. I didn't find any info about inlay hints for php. I spent only few minutes checking this out, so this might not be correct

**Reply by scippio** [reply]
btw: I found this settings for phpactor: https://phpactor.readthedocs.io/en/master/reference/configuration.html#language-server-worse-reflection-inlay-hints-enable

it works. not perfectly but works.
